### ISSUE TITLE
NullReferenceException in TraceInstance

### ISSUE THREAD

**Issue by sebastianburckhardt** [Azure/durabletask#554]
`ProcessWorkflowCompletedTaskDecision` throws a  NullReferenceException on the last line of

```csharp
TraceHelper.TraceInstance(
                TraceEventType.Information,
                "TaskOrchestrationDispatcher-InstanceCompletionEvents",
                runtimeState.OrchestrationInstance,
                () => Utils.EscapeJson(DataConverter.Serialize(runtimeState.GetOrchestrationRuntimeStateDump().Events, true)));
```

The problem is that in release builds, the GetOrchestrationRuntimeStateDump() leaves the Events property blank:

```csharp
 return new OrchestrationRuntimeStateDump
            {
                EventCount = Events.Count,
                NewEventsCount = NewEvents.Count,
            };
```
Which then causes null to be returned by `DataConverter.Serialize`, which then throws inside `Utils.EscapeJson`.

**Comment by davidmrdavid** [comment]
I wonder if this is something we could catch with the "nullable types" refactor we've been meaning to do for a while now.

**Comment by ConnorMcMahon** [comment]
Fixed in #560
